### PR TITLE
[LinearAlgebra] Fix segfault when accessing and empty CSR matrix (through iterators)

### PR DESCRIPTION
**Pull request by courtecuisse**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/CompressedRowSparseMatrixGeneric.h [hunk 1]
```diff
@@ -254,7 +254,14 @@ public :
 
     const VecIndex& getRowIndex() const { return rowIndex; }
     const VecIndex& getRowBegin() const { return rowBegin; }
-    Range getRowRange(Index id) const { return Range(rowBegin[id], rowBegin[id+1]); }
+
+    Range getRowRange(Index id) const
+    {
+        if ((unsigned) id+1 >= rowBegin.size()) return Range();
```
Comment (alxbilger) [inline Sofa/framework/LinearAlgebra/src/sofa/linearalgebra/CompressedRowSparseMatrixGeneric.h:260]:
```suggestion
        if ((unsigned) id >= rowBegin.size()) return Range();
```
`id == rowBegin.size()` would also lead to an out-of-bounds.

**Review by alxbilger — CHANGES_REQUESTED** [review]
The change makes sense, but I would like to know if your code worked in the past, before some SOFA changes, as it appears that the method `getRowRange` never checked the bounds.

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
Well spotted @courtecuisse 

- This fix reminds me that some bound checks (e.g type::vector) does only check in DEBUG mode/assertion https://github.com/sofa-framework/sofa/blob/649a41bf386b78c95264d9a6fb3125f4662c5a58/Sofa/framework/Type/src/sofa/type/vector_T.h#L108
obviously for performance reason. Should we do that here ? Thing is, many people do not run in debug so.... 🤐

- I guess this is a far cry but @fjourdes  you would want to get the fix ? (if you dont have it already...)

**Comment by fredroy** [comment]
Superseded by https://github.com/sofa-framework/sofa/pull/4236
(@courtecuisse)

**Pull request closed without merging**